scripts/gpu_drivers.sh

```diff
@@ -1,9 +1,19 @@
 #!/bin/bash
 
+# Disable popups to restart services after `apt-get upgrade` by restarting them automatically
+sudo sed -i "s/\#\$nrconf{restart} = 'i';/\$nrconf{restart} = 'a';/" /etc/needrestart/needrestart.conf
+# Don't install suggested or recommended packages with `apt`
+printf "APT::Install-Suggests "0";\nAPT::Install-Recommends "0";" >> /etc/apt/apt.conf.d/99Recommended
+
 sudo apt-get update && sudo apt-get upgrade -y
+# Pin kernel version to prevent driver/kernel mismatches
+sudo apt-mark hold linux-generic linux-image-generic linux-headers-generic
```

Comment:
I'd like to check whether the `linux-generic` virtual package doesn't already contain `linux-image-generic` but hopefully you've done that alredy. I think it's also worth pinning `linux-extras` if installed and not already included by these virtual packages.

Reply:
Great point, from the below I believe it is enough to pin `linux-generic`, as `linux-extras` has been renamed to `linux-modules-extra-<VERSION>-generic` for Ubuntu 22.04.
```
$ apt-cache depends linux-generic
linux-generic
  Depends: linux-image-generic
  Depends: linux-headers-generic
$ apt-cache depends linux-image-generic
linux-image-generic
  Depends: linux-image-5.15.0-91-generic
  Depends: linux-modules-extra-5.15.0-91-generic
  Depends: linux-firmware
  Depends: intel-microcode
  Depends: amd64-microcode
  Recommends: thermald
$ apt-cache depends linux-headers-generic
linux-headers-generic
  Depends: linux-headers-5.15.0-91-generic
```